### PR TITLE
Azeema added, Bushra wireframe included

### PR DESCRIPTION
**Pull request by kwikadi**
Two versions of azeema, stream and nonstream.

### CODE REVIEW

File: bushra.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import sendgrid
+from datetime import date, timedelta
+import ideas
+
+sendgrid = sendgrid.SendGridClient(
+    config["SendGrid"]["API_User"],
```
Comment (dufferzafar) [inline bushra.py:7]: config is not defined? `from db import config`, I guess.

**Comment by dufferzafar** [comment]
The streaming version won't actually be running right?

File: azeema-nonstream.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+Tweet = namedtuple('Tweet', 'id, msg')
+
+LIMIT = int(db.config.get("Constants","Tweets_Per_Hour")) / 2
+
+data = twitter.search(q='@IdeabinBot', since_id=last_tweet_id)
+
+for count,x in enumerate(data['statuses']):
+
+  if count == LIMIT:
+    break
```
Comment (dufferzafar) [inline azeema-nonstream.py]: Inconsistent indentation!

File: db.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 config = ConfigParser()
 config.read(os.path.join(CURRENT_DIR, 'config.ini'))
 
-DATABASE = config["Database"]["File"]
+DATABASE = config.get("Database","File")
```
Comment (dufferzafar) [inline db.py]: Is there a reason we're preferring get? I think it'd only make sense if we already have some default value specified, which we don't. So this, won't err even if no db file is defined in the config (but It should right?)

File: db.py [hunk 4]
```diff
@@ -84,4 +84,21 @@ def create_tables(con):
...
+        CREATE TABLE IF NOT EXISTS id_value (
+            tweet_id    INT ( 20 )   PRIMARY KEY
+        );
+        """
+    )
+
+    cur.execute(
+        """
+        CREATE TABLE IF NOT EXISTS retweets (
+            retweetid   INT( 20 )   UNIQUE,
```
Comment (dufferzafar) [inline db.py]: since we have `tweet_id` above, perhaps rename this to `retweet_id`?

File: tweets.py [hunk 5]
```diff
@@ -25,10 +26,12 @@ def new(idea):
...
     else:
         msg = "New idea: "
-
-    msg = tweetify(msg, idea.description, GIST_URL.format(idea.gistid))
-    tweet = twitter.update_status(status=msg)
-    return Tweet(tweet['id'], msg)
+    try:
+        msg = tweetify(msg, idea.description, GIST_URL.format(idea.gistid))
+        tweet = twitter.update_status(status=msg)
+    except:
```
Comment (dufferzafar) [inline tweets.py]: Bare excepts, no no! (mention the exact exception we're catching)

File: tweets.py [hunk 6]
```diff
@@ -1,15 +1,16 @@
 """ Tweets out an idea's descriptions. """
 
 import users
+import db
```
Comment (dufferzafar) [inline tweets.py]: Unused import? Try [this](https://github.com/DamnWidget/anaconda/).

File: db.py (hunk 3 above)
Comment (kwikadi) [inline db.py]: This is python2 stuff that I guess I missed. Fix upcoming

**Comment by dufferzafar** [comment]
I'll have to [squash](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) all your shit man.

**Comment by kwikadi** [comment]
Take a look now, fixed most of it, I think

File: ideas.py [hunk 7]
```diff
@@ -49,8 +57,8 @@ def fetch_gists(user):
 
     url = GISTS_URL.format(user.username)
     params = {
-        'client_id': db.config["Github"]["Client_Id"],
-        'client_secret': db.config["Github"]["Client_Secret"]
+        'client_id': db.config.get("Github","Client_Id"),
```
Comment (dufferzafar) [inline ideas.py]: same issue with these get calls.

**Comment by dufferzafar** [comment]
Since we don't have any defaults, I think get doesn't make much sense.

**Pull request merged**